druid/src/app_delegate.rs

```diff
@@ -57,46 +55,23 @@ type EventFn<T> = dyn Fn(Event, &mut T, &Env, &mut DelegateCtx) -> Option<Event>
 /// It is a natural place for things like window and menu management.
 ///
 /// You customize the `AppDelegate` by passing closures during creation.
-pub struct AppDelegate<T> {
-    event_fn: Option<Box<EventFn<T>>>,
-}
-
-impl<T: Data> AppDelegate<T> {
-    /// Create a new `AppDelegate`.
-    pub fn new() -> Self {
-        AppDelegate { event_fn: None }
-    }
-
-    /// Set the `AppDelegate`'s event handler. This function receives all events,
+pub trait AppDelegate<T: Data> {
+    /// The `AppDelegate`'s event handler. This function receives all events,
     /// before they are passed down the tree.
     ///
     /// The return value of this function will be passed down the tree. This can
     /// be the even that was passed in, a different event, or no event. In all cases,
     /// the `update` method will be called as usual.
-    pub fn event_handler<F>(mut self, f: F) -> Self
-    where
-        F: Fn(Event, &mut T, &Env, &mut DelegateCtx) -> Option<Event> + 'static,
-    {
-        self.event_fn = Some(Box::new(f));
-        self
-    }
-
-    pub(crate) fn event(
+    fn event(
         &mut self,
         event: Event,
         data: &mut T,
         env: &Env,
         ctx: &mut DelegateCtx,
     ) -> Option<Event> {
-        match self.event_fn.as_ref() {
-            Some(f) => (f)(event, data, env, ctx),
-            None => Some(event),
-        }
-    }
-}
-
-impl<T: Data> Default for AppDelegate<T> {
-    fn default() -> Self {
-        AppDelegate::new()
+        let _ = data;
+        let _ = env;
+        let _ = ctx;
```

Comment:
what's this about?

Reply:
oh, unused warnings? I would annotate this function `#[allow(unused)]`.

Reply:
Yeah, I was following the discussion mentioned in https://github.com/xi-editor/druid/pull/283#discussion_r344462460, and it seemed to be the most correct solution, but the attribute is also fine with me. 😄

Reply:
Done!

Reply:
ah gotcha, missed that comment. We have adopted the pattern of using `#[allow]` on each function with this problem, so let's just keep doing that. 🤷‍♂